src/coreclr/src/ilasm/grammar_after.cpp

```diff
@@ -78,7 +78,11 @@ static Keywords keywords[] = {
 /*--------------------------------------------------------------------------*/
 char* nextcharA(__in __nullterminated char* pos)
 {
+#ifdef HOST_WINDOWS
     return (*pos > 0) ? ++pos : (char *)_mbsinc((const unsigned char *)pos);
+#else
+    return ++pos;
```

Comment:
Looks like the only usage of `nextcharA` is wrong as it is comparing function pointers instead of result of functions, so this condition always returns true:
https://github.com/dotnet/runtime/blob/342d9d342a440121def32c09735881ba64bd0cdc/src/coreclr/src/ilasm/grammar_after.cpp#L938

Reply:
Yes, this is comparing function pointers; but I do not see why this condition would be always true.

Reply:
I was thinking it should do `nextchar(curPos) != nextcharA(curPos)`.

Reply:
The code is using the function pointer as encoding identifier. `nextchar(curPos) != nextcharA(curPos)` would change what this code does.

Reply:
Ah ok, I just saw the definition of `nextchar`, it is not a function just a handle declared elsewhere which changes contextually.